File: 4.Group Anagrams/Solution.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    if(strs.length==0){
        return [[]];
    }

    const ansMap = new Map();

    for(s of strs){
        const count = new Array(26).fill(0);

        for(c of s){
            count[c.charCodeAt(0)-'a'.charCodeAt(0)]++;
        }

        let sb = "";

        for(let i=0;i<26;i++){
            sb+="#";
            sb+=String(count[i]);
        }

        if(!ansMap.has(sb)){
            ansMap.set(sb,[]);
        }

        ansMap.get(sb).push(s);
    }

    return Array.from(ansMap.values());
};

File: 4.Group Anagrams/Solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if len(strs)==0:
            return [[]]
        ansMap = {}

        

        for s in strs:
            count = [0]*26
            for c in s:
                count[ord(c)-ord('a')]+=1
        
            key = ""

            for i in count:
                key+='#'
                key+=str(i)

            if key not in ansMap.keys():
                ansMap[key] = []

            ansMap[key].append(s)

        return [l for l in ansMap.values()] 

        